acceptance/tests/puppet_device_test.rb

```diff
@@ -1,74 +1,73 @@
+  }
+
+  # install and configure the Resource API
+  include 'resource_api::server', 'resource_api::agent'
 MANIFEST
 
-on(master, "cp -v #{confdir}/puppet.conf /tmp/puppet.conf")
-on(master, "echo \"#{<<-CONF}\" >> /tmp/puppet.conf")
-[main]
-user = test
+  on(master, "echo \"#{<<-CONF}\" >> #{confdir}/puppet.conf")
+  [main]
+  user = test
 CONF
 
-# configure device credentials
-on(master, puppet("apply -e 'device_manager { [\"spinny1.example.com\", \"spinny2.example.com\", \"spinny3.example.com\"]: type => 'spinner', credentials => { facts_cpu_time => 1 }, include_module => false }'"))
+  # configure device credentials
+  on(master, puppet("apply -e 'device_manager { [\"spinny1.example.com\", \"spinny2.example.com\", \"spinny3.example.com\"]: type => 'spinner', credentials => { facts_cpu_time => 1 }, include_module => false }'"))
 
-# run operation without certificate
-on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com --facts")
+  # run operation without certificate
+  on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com --facts")
 
-# request cert, or fall through
-on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com", acceptable_exit_codes: [0, 1]) do |result|
-  assert_no_match(/Permission denied/, result.stderr, 'cert requesting failed')
-end
+  # request cert, or fall through
+  on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com", acceptable_exit_codes: [0, 1]) do |result|
+    assert_no_match(/Permission denied/, result.stderr, 'cert requesting failed')
+  end
 
-on(master, "umask 077; puppetserver ca sign --certname spinny1.example.com")
+  on(master, "umask 077; puppetserver ca sign --certname spinny1.example.com")
 
-# test catalog application
-on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com")
+  # test catalog application
+  on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com")
 
-# test --resource
-on(master, "umask 077; puppet device  #{common_options} --target spinny1.example.com --resource spinner")
+  # test --resource
+  on(master, "umask 077; puppet device  #{common_options} --target spinny1.example.com --resource spinner")
 
-# test --apply
-on(master, "umask 077; puppet device  #{common_options} --target spinny1.example.com --apply /tmp/spinner.pp")
+  # test --apply
+  on(master, "umask 077; puppet device  #{common_options} --target spinny1.example.com --apply /tmp/spinner.pp")
 
-# test development runmode
-common_options += " --libdir #{codedir}/test_device/lib"
-on(master, "umask 077; puppet device #{common_options} --target spinny2.example.com --facts")
-on(master, "umask 077; puppet device #{common_options} --target spinny2.example.com", acceptable_exit_codes: [0, 1]) do |result|
-  assert_no_match(/Permission denied/, result.stderr, 'cert requesting failed with --libdir')
+  # test development runmode
```

Comment:
what does this comment mean?

Reply:
@DavidS ?

Reply:
@tphoney ?

Reply:
`puppet device` acts differently when `--libdir` is specified. Therefore I added a separate call with it here.